examples/02-plot/spherical.py

```diff
@@ -0,0 +1,226 @@
+"""
+Plot data in spherical coordinates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generate and visualize meshes from data in longitude-latitude coordinates.
+"""
+
+import pyvista as pv
+import numpy as np
+
+
+def _cell_bounds(points, bound_position=0.5):
+    """
+    Calculate coordinate cell boundaries.
+
+    Parameters
+    ----------
+    points: numpy.array
+        One-dimensional array of uniformy spaced values of shape (M,)
+    bound_position: bool, optional
+        The desired position of the bounds relative to the position
+        of the points.
+
+    Returns
+    -------
+    bounds: numpy.array
+        Array of shape (M+1,)
+
+    Examples
+    --------
+    >>> a = np.arange(-1, 2.5, 0.5)
+    >>> a
+    array([-1. , -0.5,  0. ,  0.5,  1. ,  1.5,  2. ])
+    >>> cell_bounds(a)
+    array([-1.25, -0.75, -0.25,  0.25,  0.75,  1.25,  1.75,  2.25])
+    """
+    assert points.ndim == 1, "Only 1D points are allowed"
+    diffs = np.diff(points)
+    delta = diffs[0] * bound_position
+    bounds = np.concatenate([[points[0] - delta], points + delta])
+    return bounds
+
+
+def grid_from_sph_coords(lons, lats, levels):
+    """
+    Create a structured grid from arrays of spherical coordinates.
+
+    Parameters
+    ----------
+    lon: array-like
+        Array of longitudes of shape (M,) [degrees]
+    lat: array-like
+        Array of latitudes of shape (N,) [degrees]
+    levels: array-like
+        Array of vertical levels of shape (P,)
+
+    Returns
+    -------
+    pyvista.StructuredGrid
+    """
+    x, y, z = np.meshgrid(np.radians(lons), np.radians(lats), levels)
+    # Transform grid to cartesian coordinates
+    x_cart = levels * np.cos(y) * np.cos(x)
+    y_cart = levels * np.cos(y) * np.sin(x)
+    z_cart = levels * np.sin(y)
+    # Make a grid object
+    return pv.StructuredGrid(x_cart, y_cart, z_cart)
```

Comment:
This function seems highly useful and general. Perhaps this should go in `pyvista/utilities/features.py`